bittide/src/Bittide/ScatterGather.hs

```diff
@@ -22,12 +22,16 @@ import Bittide.Calendar
 import Bittide.DoubleBufferedRam
 import Bittide.SharedTypes
 
+-- | GADT to explicitly differentiate between a configuration for the 'scatterUnitWb' and
```

Comment:
This is actually a GADT, but an existential type, that just happens to be defined using [GADTSyntax](https://downloads.haskell.org/ghc/9.2.4/docs/html/users_guide/exts/gadt_syntax.html#extension-GADTSyntax)

Reply:
But you can probably leave those details out, if what you care about documenting here is that they allow you differentiate between scatter and gather config.
The existential part is what allows you to hide the `memDepth`.